src/main/java/seedu/address/logic/commands/CommandWord.java

```diff
@@ -39,16 +41,31 @@
             inputs.put(commandWord.inputString, commandWord);
         }
     }
+
     private final String inputString;
+    private final List<CommandType> validTypes;
 
-    CommandWord(String inputString) {
+    CommandWord(String inputString, List<CommandType> validTypes) {
         this.inputString = inputString;
+        this.validTypes = new ArrayList<>(validTypes);
     }
 
     public static CommandWord get(String inputString) {
         return Optional.ofNullable(inputs.get(inputString)).orElse(DEFAULT);
     }
 
+    public boolean requiresType() {
+        return !validTypes.isEmpty();
+    }
+
+    public String acceptedTypes() {
+        return validTypes.stream().map(CommandType::getTypeString).reduce("", (x, y) -> x + "\n" + y);
+    }
```

Comment:
perhaps this should return a list of accepted `CommandType`s? Otherwise, I think the name is slightly misleading. Maybe something like `getAcceptedTypesString() {..}`

Reply:
good call.